config/components/prometheus/kustomization.yaml

```diff
@@ -1,3 +1,11 @@
 resources:
 - monitor.yaml
 - role.yaml
+# [PROMETHEUS-WITH-CERTS] The following patch configures the ServiceMonitor in ../prometheus
+# to securely reference certificates created and managed by cert-manager.
+# Additionally, ensure that you uncomment the [METRICS WITH CERTMANAGER] patch under config/default/kustomization.yaml
+# to mount the "metrics-server-cert" secret in the Manager Deployment.
+patches:
+- path: monitor_tls_patch.yaml
+  target:
+    kind: ServiceMonitor
```

Comment:
Why do we need to apply TLS patch by default?

Reply:
yeah, we want to make sure the Prometheus config works as before.

Reply:
This is still remaining.

Reply:
I think we can still kept this based on @mimowo comments.

if I remove this it breaks the ability to configure tlsConfigs.

Reply:
Actually, I agree with @tenzen-y  - i missed this. We should not change the default configuration, so I suggest the patch to be opt-in.